### PR TITLE
Fix TinyUSB HID polling, add debug macro

### PR DESCRIPTION
**Pull request by Spegs21**

### CODE REVIEW

**Comment by Spegs21** [comment]
@Legion2 This is ready now.

**Review by Legion2 — CHANGES_REQUESTED** [review]
Thanks for the fix and the debug tools

File: src/CLPUtils.cpp [hunk 1]
```diff
@@ -66,3 +66,44 @@ void CLP::printFps(const int interval) {
...
+	va_list ap;
+	va_start(ap, format);
+	strcpy_P(fmt, (const char*)format);
+	len = vsnprintf(buf, sizeof(buf), fmt, ap);
+	CLP_DEBUG_PORT.write(buf);
+	va_end(ap);
+	return len;
+}
+
+void clp_print_data(uint8_t const* buf, uint32_t bufsize, bool address_table) {
```
[the remark this answers is not in the document]
Comment (Spegs21) [reply src/CLPUtils.cpp:96]: Sure, I'll fix it up.

[the remark this answers is not in the document]
Comment (Spegs21) [reply src/CLPUtils.cpp]: Should be good now. Lmk and I'll squash/rebase.

File: src/CLPUtils.h [hunk 2]
```diff
@@ -29,56 +29,59 @@
 #define CLP_XSTRCAT3(a, b, c) CLP_STRCAT3(a, b, c)
 
 #if defined(CLP_DEBUG_PORT)
-extern int CLP_printf(const char* format, ...);
-extern int CLP_printf(const __FlashStringHelper* format, ...);
-#define clp_printf CLP_printf
+namespace CLP {
+extern int printf(const char* format, ...);
+extern int printf(const __FlashStringHelper* format, ...);
```
Comment (Legion2) [inline src/CLPUtils.h:34]: what das the `...` mean here, I don't know this syntax

File: src/CorsairLightingProtocolTinyUSBHID.cpp [hunk 3]
```diff
@@ -26,7 +26,7 @@ uint8_t const hid_report[] = {HID_USAGE_PAGE_N (HID_USAGE_PAGE_VENDOR | 0xC0, 2)
 							  HID_COLLECTION_END};
 /* clang-format on */
 
-Adafruit_USBD_HID TinyUSBDevice_HID(hid_report, sizeof(hid_report), HID_ITF_PROTOCOL_NONE, 2, true);
+Adafruit_USBD_HID tudHid(hid_report, sizeof(hid_report), HID_ITF_PROTOCOL_NONE, 2, true);
```
Comment (Legion2) [inline src/CorsairLightingProtocolTinyUSBHID.cpp:29]: did you test with 200 and more leds? why did you choose `2` here?

File: src/CLPUtils.h (hunk 2 above)
Comment (Spegs21) [reply src/CLPUtils.h:34]: It's a special symbol for [variadic functions](https://en.cppreference.com/w/cpp/utility/variadic). It allows a variable amount of arguments which is useful for formatting with printf.

File: src/CorsairLightingProtocolTinyUSBHID.cpp (hunk 3 above)
Comment (Spegs21) [reply src/CorsairLightingProtocolTinyUSBHID.cpp:29]: Good catch! I was doing some testing and forgot to change it back.

File: src/CLPUtils.h (hunk 2 above)
Comment (Legion2) [reply src/CLPUtils.h:34]: Ok I didn't know you can put a comma there

File: src/CorsairLightingProtocolTinyUSBHID.cpp (hunk 3 above)
Comment (Legion2) [reply src/CorsairLightingProtocolTinyUSBHID.cpp]: can we set it to `0`?

Comment (Spegs21) [reply src/CorsairLightingProtocolTinyUSBHID.cpp]: I haven't tried but I don't think we need to. 25 FPS is the max iCue allows and we get that with 1.

**Review by Legion2 — APPROVED** [review]

**Pull request merged**